File: brainbox/reverse_prefix_word.py
def reversePrefix(word: str, ch: str) -> str:
    idx = word.find(ch)
    if idx == -1:
        return word
    return word[:idx+1][::-1] + word[idx+1:]

def test_reversePrefix():
    print(reversePrefix("abcdefd", "d"), "== dcbaefd")
    print(reversePrefix("hello", "l"), "== lleho")
    print(reversePrefix("hello", "z"), "== hello")
    print(reversePrefix("abc", "c"), "== cba")
    print(reversePrefix("a", "a"), "== a")
    print(reversePrefix("xyz", "x"), "== xyz")  # reversing only 'x' does nothing

test_reversePrefix()

File: brainbox/reverse_prefix_word.ts
export function reversePrefix(word: string, ch: string): string {
  const idx = word.indexOf(ch);
  if (idx === -1) return word;
  return word.slice(0, idx + 1).split("").reverse().join("") + word.slice(idx + 1);
}

import { reversePrefix } from "./reversePrefix";

function test_reversePrefix() {
  console.log(reversePrefix("abcdefd", "d"), "== dcbaefd");
  console.log(reversePrefix("hello", "l"), "== lleho");
  console.log(reversePrefix("hello", "z"), "== hello");
  console.log(reversePrefix("abc", "c"), "== cba");
  console.log(reversePrefix("a", "a"), "== a");
  console.log(reversePrefix("xyz", "x"), "== xyz"); // reversing only 'x' does nothing
}

test_reversePrefix();
